sqlglot/dialects/dialect.py

```diff
@@ -1295,7 +1295,7 @@ def trim_sql(self: Generator, expression: exp.Trim) -> str:
     collation = self.sql(expression, "collation")
 
     # Use TRIM/LTRIM/RTRIM syntax if the expression isn't database-specific
-    if not remove_chars and not collation:
```

Comment:
Curious about this term in the condition - did you find any context around it?

Reply:
I actually didn't find much regarding collation, at least for the relevant dialects such as [Oracle](https://docs.oracle.com/cd/B12037_01/server.101/b10759/functions184.htm), [Postgres](https://www.postgresql.org/docs/9.1/functions-string.html), [MySQL](https://dev.mysql.com/doc/refman/8.4/en/string-functions.html#function_trim).  We do have a Postgres test for `TRIM(... COLLATE utf8_bin)` (although this specific collation doesn't work for me locally) which I verified is a valid query. 

I think the condition there was probably not needed for Postgres; If we don't have `remove_chars` (the default char is a space), then the RTT versions are semantically equivalent:

```
Input: SELECT TRIM(LEADING ' XXX ' COLLATE "de_DE");
SQLGlot RTT: SELECT LTRIM(' XXX ' COLLATE "de_DE")


Input: SELECT TRIM(' XXX ' COLLATE "de_DE")
SQLGlot RTT: SELECT TRIM(' XXX ' COLLATE "de_DE")
```



Source PRs: https://github.com/tobymao/sqlglot/pull/432, https://github.com/tobymao/sqlglot/pull/934/files

Reply:
Cool, thanks for double checking.